marshmallow/schema.py

```diff
@@ -174,6 +201,37 @@ def _resolve_processors(self):
                 self.__processors__[tag].append(attr_name)
 
 
+class SchemaJit(with_metaclass(ABCMeta)):
```

Comment:
This has to be part of Marshmallow itself as it sets up the contract for dealing with the external library, otherwise the external library has to duplicate huge swaths of this file to inject itself.  Doing this via hooks and a contract creates a much looser coupling with Marshmallow and the external library.

The other main issue with the subclass approach is that it doesn't allow some of the other great libraries in the Marshmallow ecosystem to benefit from this.  For example there's no way for me to compose a `flask_marshmallow.Schema` with a `toastedmarshmallow.JitSchema` without effectively doubling the class hierarchy in all of those other frameworks.